processing/src/main/java/io/druid/query/groupby/orderby/OrderByColumnSpec.java

```diff
@@ -42,50 +41,74 @@
+      for (Direction direction : Direction.values()) {
+        bob.put(direction.name(), direction);
+      }
+      stupidEnumMap = bob.build();
+    }
 
-  public static final StringComparator DEFAULT_DIMENSION_ORDER = StringComparators.LEXICOGRAPHIC;
+    @JsonValue
+    @Override
+    public String toString()
+    {
+      return this.name().toLowerCase();
+    }
 
-  /**
-   * Maintain a map of the enum values so that we can just do a lookup and get a null if it doesn't exist instead
-   * of an exception thrown.
-   */
-  private static final Map<String, Direction> stupidEnumMap;
+    @JsonCreator
+    public static Direction fromString(String name)
+    {
+      final String upperName = name.toUpperCase();
+      Direction direction = stupidEnumMap.get(upperName);
+
+      if (direction == null) {
+        for (Direction dir : Direction.values()) {
+          if (dir.name().startsWith(upperName)) {
+            if (direction != null) {
+              throw new ISE("Ambiguous directions[%s] and [%s]", direction, dir);
+            }
+            direction = dir;
+          }
+        }
+      }
 
-  static {
-    final ImmutableMap.Builder<String, Direction> bob = ImmutableMap.builder();
-    for (Direction direction : Direction.values()) {
-      bob.put(direction.toString(), direction);
+      return direction;
     }
-    stupidEnumMap = bob.build();
   }
 
+  public static final StringComparator DEFAULT_DIMENSION_ORDER = StringComparators.LEXICOGRAPHIC;
+
   private final String dimension;
   private final Direction direction;
   private final StringComparator dimensionComparator;
 
   @JsonCreator
-  public static OrderByColumnSpec create(Object obj)
+  public OrderByColumnSpec(
+      @JsonProperty("dimension") String dimension,
+      @JsonProperty("direction") Direction direction,
+      @JsonProperty("dimensionOrder") StringComparator dimensionComparator
+  )
   {
-    Preconditions.checkNotNull(obj, "Cannot build an OrderByColumnSpec from a null object.");
-
-    if (obj instanceof String) {
-      return new OrderByColumnSpec(obj.toString(), null, null);
-    } else if (obj instanceof Map) {
-      final Map map = (Map) obj;
-
-      final String dimension = map.get("dimension").toString();
-      final Direction direction = determineDirection(map.get("direction"));
-      final StringComparator dimensionComparator = determinDimensionComparator(map.get("dimensionOrder"));
+    this.dimension = dimension;
+    this.direction = direction == null ? Direction.ASCENDING : direction;
+    this.dimensionComparator = dimensionComparator == null ? DEFAULT_DIMENSION_ORDER : dimensionComparator;
+  }
 
-      return new OrderByColumnSpec(dimension, direction, dimensionComparator);
-    } else {
-      throw new ISE("Cannot build an OrderByColumnSpec from a %s", obj.getClass());
-    }
+  @JsonCreator
+  public static OrderByColumnSpec fromString(String dimension)
```

Comment:
Are there serde tests for this class? Having two `@JsonCreator` on the same class seems like the kind of thing that might work or might not work.